common/flatpak-dir.c

```diff
@@ -11112,6 +11112,53 @@ flatpak_dir_get_deployed_metadata (FlatpakDir    *self,
   return TRUE;
 }
 
+/* Get the metadata the most recent commit of ref in the
+ * repo. This is not the same as the xa.cache, which might
+ * be updated separately. This function is used by
+ * flatpak_dir_find_local_related_for_ref_in_repo because we
+ * want to find the local related refs for a ref that has
+ * been pulled into the repo already, not for a ref for which
+ * an updated piece of metadata might exist on the remote end
+ * that we pulled into the xa.cache through some other means. */
+static GKeyFile *
+flatpak_dir_get_metadata_for_ref_in_repo (FlatpakDir    *self,
+                                          const char    *ref,
+                                          const char    *remote_name,
+                                          GCancellable  *cancellable,
+                                          GError       **error)
+{
+  g_autofree char *refspec = g_strdup_printf ("%s:%s", remote_name, ref);
+  g_autofree char *checksum = NULL;
+  g_autoptr(GVariant) commit = NULL;
+  g_autoptr(GVariant) metadata = NULL;
+  const char *xa_metadata_str = NULL;
+  g_autoptr(GKeyFile) metakey = NULL;
+
+  if (!ostree_repo_open (self->repo, cancellable, error))
```

Comment:
You're supposed to use flatpak_dir_ensure_repo().

Reply:
Indeed - this was a mistake made by copy-pasting some of the logic I had in our own updater.